pkg/provisioning/service.go

```diff
@@ -43,25 +48,63 @@ func NewService(
 	plService PipelineService,
 	connService ConnectorService,
 	procService ProcessorService,
+	pipelinesDir string,
 ) *Service {
 	return &Service{
 		db:               db,
 		logger:           logger.WithComponent("provisioning.Service"),
 		pipelineService:  plService,
 		connectorService: connService,
 		processorService: procService,
+		pipelinesPath:    pipelinesDir,
 	}
 }
 
-func (s *Service) ProvisionConfigFile(ctx context.Context, path string) ([]string, error) {
+func (s *Service) Init(ctx context.Context) error {
+	s.logger.Debug(ctx).Str("pipelinesDir", s.pipelinesPath).Msg("initializing the provisioning service")
+
+	var files []string
+	err := filepath.WalkDir(s.pipelinesPath, func(path string, fileInfo fs.DirEntry, err error) error {
+		if strings.HasSuffix(path, ".yml") {
```

Comment:
I wonder why not use one yaml file rather than multiple? Won't that make processing simpler?

Reply:
We support both approaches, it's your choice which one you take. Once you reach a certain number of pipelines it will be nicer to manage them in separate files IMO.